cache/lib/cache_web/controllers/module_cache_controller.ex

```diff
@@ -0,0 +1,331 @@
+defmodule CacheWeb.ModuleCacheController do
+  use CacheWeb, :controller
+  use OpenApiSpex.ControllerSpecs
+
+  alias Cache.BodyReader
+  alias Cache.CacheArtifacts
+  alias Cache.Disk
+  alias Cache.S3
+  alias Cache.S3Transfers
+  alias CacheWeb.API.Schemas.Error
+
+  require Logger
+
+  plug OpenApiSpex.Plug.CastAndValidate, json_render_error_v2: true
+
+  tags(["ModuleCache"])
+
+  operation(:download,
+    summary: "Download a module cache artifact",
+    operation_id: "downloadModuleCacheArtifact",
+    parameters: [
+      project_id: [
+        in: :query,
+        type: :string,
+        required: true,
+        description: "Project ID (account_handle/project_handle)"
+      ],
+      hash: [
+        in: :query,
+        type: :string,
+        required: true,
+        description: "Artifact hash"
+      ],
+      name: [
+        in: :query,
+        type: :string,
+        required: true,
+        description: "Artifact name"
+      ],
+      cache_category: [
+        in: :query,
+        type: :string,
+        required: false,
+        description: "Cache category (builds, selective_tests)"
+      ]
```

Comment:
the selective tests are stored in the db and shouldn't be stored in the cache node.